mpas_analysis/configuration/MpasAnalysisConfigParser.py

```diff
@@ -13,44 +35,24 @@ def getWithDefault(self, section, option, default, listType=None):
                     return self.getfloat(section, option)
                 elif isinstance(default, bool):
                     return self.getboolean(section, option)
-                elif isinstance(default, list):
-                    if (listType is None) and (len(default) > 0):
-                        # use the type of default[0] for listType
-                        listType = type(default[0])
-
-                    return self.getlist(section, option, listType)
+                elif isinstance(default, (list, tuple, dict)):
+                    return self.getExpression(section, option)
                 else:
                     return self.get(section, option)
 
         # we didn't find the entry so set it to the default
-        if isinstance(default, list):
-            self.setlist(section, option, default)
-        else:
-            self.set(section, option, str(default))
+        self.set(section, option, str(default))
         return default
 
-    def getlist(self, section, option, listType=str):
-        listString = self.get(section, option)
-        # first, strip the brackets
-        listString = listString.strip('[]')
-        if (listType == str) and ('"' in listString):
-            # assume all entries are strings separated by quotes,
-            # but every other one is the comma separating list entries
-            stringList = listString.split('"')[0::2]
-        elif (listType == str) and ("'" in listString):
-            # assume all entries are strings separated by quotes,
-            # but every other one is the comma separating list entries
-            stringList = listString.split("'")[0::2]
-        else:
-            stringList = listString.split(',')
-        outList = []
-        for string in stringList:
-            outList.append(listType(string.strip()))
-        return outList
-
-    def setlist(self, section, option, value):
-        listString = ', '.join(str(entry) for entry in value)
-        listString = '[%s]'%listString
-        self.set(section, option, listString)
-
-
+    def getExpression(self, section, option):
+        """
+        Get an option as an expression (typically a list, though tuples and
+        dicts should also work).  `section` and `option` work as in `get(...)`.
+        The expression is required to have valid python syntax, so that
+        string entries are required to be in single or double quotes.
+
+        Author: Xylar Asay-Davis
+        Last Modified: 12/03/2016
+        """
+        expressionString = self.get(section, option)
+        return ast.literal_eval(expressionString)
```

Comment:
@xylar, very nice!!!  This removes a ton of code and is a really clean solution.